streamer/src/nonblocking/quic.rs

```diff
@@ -519,10 +565,76 @@ async fn setup_connection(
     }
 }
 
+async fn packet_batch_sender(
+    packet_sender: Sender<PacketBatch>,
+    packet_receiver: AsyncReceiver<PacketAccumulator>,
+    exit: Arc<AtomicBool>,
+    stats: Arc<StreamStats>,
+    coalesce_ms: u64,
+) {
+    trace!("enter packet_batch_sender");
+    let coalesce_ms = coalesce_ms as u128;
+    let mut batch_start_time = Instant::now();
+    loop {
+        let mut packet_batch = PacketBatch::with_capacity(PACKETS_PER_BATCH);
+
+        stats
+            .total_packets_allocated
+            .fetch_add(PACKETS_PER_BATCH, Ordering::Relaxed);
+        loop {
+            if exit.load(Ordering::Relaxed) {
+                return;
+            }
+            let elapsed = batch_start_time.elapsed();
+            if packet_batch.len() >= PACKETS_PER_BATCH
+                || (!packet_batch.is_empty() && elapsed.as_millis() >= coalesce_ms)
+            {
+                let len = packet_batch.len();
+                if let Err(e) = packet_sender.send(packet_batch) {
+                    stats
+                        .total_packet_batch_send_err
+                        .fetch_add(1, Ordering::Relaxed);
+                    trace!("Send error: {}", e);
```

Comment:
If receiver has been dropped, it looks like it will never succeed sending and it is safe to return (?)